Gems/GameStateSamples/CMakeLists.txt

```diff
@@ -7,3 +7,4 @@
 #
 
 add_subdirectory(Code)
+ly_create_alias(NAME GameStateSamples.Builders NAMESPACE Gem)
```

Comment:
The .Builders variants are actually only available on host platforms that build the AssetProcessor such as Windows, Linux and Mac.

Therefore variant isn't available on iOS and Android.

Because this is a Code Gem and not an Asset-Only Gem, I suggest moving this line to the [Gems/GameStateSamples/Code/CMakeLists.txt](https://github.com/o3de/o3de/blob/2e12d44e2dadda30d7d7325bfd77e02c626d6eab/Gems/GameStateSamples/Code/CMakeLists.txt) where it can placed under an `if(PAL_TRAIT_BUILD_HOST_TOOLS)` block to only be added when building host platforms.

```
if(PAL_TRAIT_BUILD_HOST_TOOLS)
    ly_create_alias(NAME GameStateSamples.Builders NAMESPACE Gem)
endif()

Reply:
I don't see a line with that if statement. Should I paste the entire if statement to the CMakeLists.txt file?

Reply:
Yes, you can place that at the bottom of that CMakeLists.txt

Reply:
So I have updated the changes to my forked repo. Does the change sync with this pull request?

Reply:
Done, please take a look

Reply:
DCO says I haven't signed off but I did :(